src/v/kafka/server/handlers/fetch.h

```diff
@@ -58,7 +62,11 @@ struct op_context {
     op_context(const op_context&) = delete;
     op_context& operator=(const op_context&) = delete;
     op_context& operator=(op_context&&) = delete;
-    ~op_context() noexcept = default;
+    ~op_context() {
+        iteration_order.clear_and_dispose([](response_placeholder* ph) {
+            delete ph; // NOLINT
```

Comment:
nit: do we not have to worry about deleting a nullptr here?

Reply:
not really, when `response_placeholder` is deleted it is automatically unliked and will no longer be in the list